blackbox/handlers/storage/dropbox.py

```diff
@@ -75,28 +76,35 @@ def sync(self, file_path: Path) -> None:
             self.success = False
             self.output = str(e)
 
-    def rotate(self) -> None:
+    def rotate(self, database_id: str) -> None:
         """
         Rotate the files in the Dropbox directory.
 
         All files in base directory of backups will be deleted when they
         are older than `retention_days`, and because of this,
         it's better to have backups in isolated folder.
         """
+        # Let's rotate only this type of database
+        db_type_regex = rf"{database_id}_blackbox_\d{{2}}_\d{{2}}_\d{{4}}.+"
+
         # Receive first batch of files.
         files_result = self.client.files_list_folder(
             self.upload_base if self.upload_base != "/" else ""
         )
         entries = [
-            entry for entry in files_result.entries if isinstance(entry, FileMetadata)
+            entry for entry in files_result.entries if
+            isinstance(entry, FileMetadata) and re.match(db_type_regex,
+                                                         entry.name)
         ]
```

Comment:
this list comprehension and the list comprehension below have grown too complex in this commit. A good rule of thumb is that a list comprehension should only do _two things_. Before this change, we did two things. We looped over `files_result.entries`, and we checked the object type. Now we're doing _three things_, which is too many.

At this point, we need to simplify the expression somehow. You have two good choices for how to do this. One is to create a helper function, like `_is_backup_file() -> bool` which you can call in the list comprehension
```py
entries = [entry for entry in files_result.entries if self._is_backup_file(entry)]
```

The other option is to just stop using a list comprehension and write a boring old for loop here instead.

```py
entries = []
for entry in files_result.entries:
    if isinstance(entry, FileMetadata) and re.match(db_type_regex, entry.name):
        entries.append(entry)
```

It's not as sexy, but it's more readable because we're doing fewer things at a time. Remember what [The Zen of Python](http://www.python.org/dev/peps/pep-0020/) says:

```
Simple is better than complex.
Complex is better than complicated.
Flat is better than nested.
Sparse is better than dense.
```